src/pages/Routes/GoogleRedirectRoute.jsx

```diff
@@ -0,0 +1,40 @@
+import axios from 'axios';
+import React, { useContext, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { AuthContext } from '../../context/authContext';
+import LoadingComponent from '../../components/Loading/LoadingComponent';
+
+function GoogleRedirectRoute() {
+  const navigate = useNavigate();
+  const { setJwt } = useContext(AuthContext);
+
+  useEffect(() => {
+    try {
+      const accessToken = window.location.hash.match(/#access_token=([^&]+)/)[1];
+      if (!accessToken) {
+        navigate('/login', { replace: true });
+      }
+
+      axios
+        .post(`${process.env.REACT_APP_BACKEND_ENDPOINT}/api/users/login`, null, {
+          headers: {
+            id_token: accessToken,
+          },
+        })
+        .then(({ data }) => {
+          setJwt(data);
+        })
+        .catch(() => {
+          setJwt('');
+        });
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
+      navigate('/login', { replace: true });
+    }
+  }, []);
+
+  return <LoadingComponent />;
+}
+
+export default GoogleRedirectRoute;
```

Comment:
Added for Google OAuth2 redirect.

set google oauth redirect to http://localhost:3000/google-oauth
e.g http://localhost:3000/google-oauth#access_token=.............................

auto parse access token and send to backend and get new jwt.